sdks/typescript/client/src/components/AppFrame.tsx

```diff
@@ -0,0 +1,253 @@
+
+  const [iframeReady, setIframeReady] = useState(false);
+  const [bridgeConnected, setBridgeConnected] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const iframeRef = useRef<HTMLIFrameElement | null>(null);
+
+  // Refs for callbacks to avoid effect re-runs
+  const onSizeChangedRef = useRef(onSizeChanged);
+  const onLoggingMessageRef = useRef(onLoggingMessage);
+  const onInitializedRef = useRef(onInitialized);
+  const onErrorRef = useRef(onError);
+
+  useEffect(() => {
+    onSizeChangedRef.current = onSizeChanged;
+    onLoggingMessageRef.current = onLoggingMessage;
+    onInitializedRef.current = onInitialized;
+    onErrorRef.current = onError;
+  });
+
+  // Effect 1: Set up sandbox iframe and connect AppBridge
+  useEffect(() => {
+    let mounted = true;
+
+    const setup = async () => {
+      try {
+        const { iframe, onReady } = await setupSandboxProxyIframe(sandbox.url);
+
+        if (!mounted) return;
+
+        iframeRef.current = iframe;
+        if (containerRef.current) {
+          containerRef.current.appendChild(iframe);
+        }
+
+        await onReady;
+
+        if (!mounted) return;
+
+        // Register size change handler
+        appBridge.onsizechange = async (params) => {
+          onSizeChangedRef.current?.(params);
+          // Also update iframe size
+          if (iframeRef.current) {
+            if (params.width !== undefined) {
+              iframeRef.current.style.width = `${params.width}px`;
+            }
+            if (params.height !== undefined) {
+              iframeRef.current.style.height = `${params.height}px`;
+            }
+          }
+        };
+
+        // Hook into initialization
+        appBridge.oninitialized = () => {
+          if (!mounted) return;
+          console.log('[AppFrame] App initialized');
+          setIframeReady(true);
+          onInitializedRef.current?.({
+            appVersion: appBridge.getAppVersion(),
+            appCapabilities: appBridge.getAppCapabilities(),
+          });
+        };
+
+        // Register logging handler
+        appBridge.onloggingmessage = (params) => {
+          onLoggingMessageRef.current?.(params);
+        };
+
+        // Connect the bridge
+        await appBridge.connect(
+          new PostMessageTransport(iframe.contentWindow!, iframe.contentWindow!),
+        );
+
+        if (!mounted) return;
+
+        setBridgeConnected(true);
+      } catch (err) {
+        console.error('[AppFrame] Error:', err);
+        if (!mounted) return;
+        const error = err instanceof Error ? err : new Error(String(err));
+        setError(error);
+        onErrorRef.current?.(error);
+      }
+    };
+
+    setup();
+
+    return () => {
+      mounted = false;
+      if (iframeRef.current && containerRef.current?.contains(iframeRef.current)) {
+        containerRef.current.removeChild(iframeRef.current);
+      }
```

Comment:
```suggestion

```

@ochafik found a lifecycle problem. This code destroys the iframe when the component unmounts. but then the mounting process doesn't happen again (create iframe setHtml). I don't think that's what we want anyway?

The iframe should preserve the state when rerendering instead of destroying and recreating the iframe?